.github/workflows/publish.yml

```diff
@@ -61,8 +61,56 @@ jobs:
       - uses: actions/setup-go@v5
         with:
           go-version: 1.22.x
+      - name: Set up Java
+        uses: actions/setup-java@v4
+        with:
+          java-version: 17
+          distribution: 'temurin'
       - name: Build binary
         run: GOOS=${{ matrix.os }} GOARCH=${{ matrix.arch }} make build
+      - name: Download jsign.jar
+        if: matrix.os == 'windows'
+        run: |
+          curl -L -o jsign.jar https://github.com/ebourg/jsign/releases/download/6.0/jsign-6.0.jar
+      - name: Verify jsign.jar
+        if: matrix.os == 'windows'
+        run: |
+          echo '05ca18d4ab7b8c2183289b5378d32860f0ea0f3bdab1f1b8cae5894fb225fa8a  jsign.jar' | sha256sum -c
+      - name: Prepare certificate chain
+        if: matrix.os == 'windows'
+        run: |
+          echo "${{ secrets.CERTIFICATE_CHAIN }}" | base64 --decode > codesign-chain.pem
+      - id: 'auth'
+        name: Authenticate with Google Cloud
+        if: matrix.os == 'windows'
+        uses: 'google-github-actions/auth@6fc4af4b145ae7821d527454aa9bd537d1f2dc5f'
+        with:
+          credentials_json: '${{ secrets.CERTIFICATE_SA_CREDENTIALS }}'
+      - name: Set up Cloud SDK
+        if: matrix.os == 'windows'
+        uses: 'google-github-actions/setup-gcloud@6189d56e4096ee891640bb02ac264be376592d6a'
```

Comment:
We usually just use the major versions. `@v2` is good for this one and auth above.